cypress/e2e/po/pages/extensions-compatibility-tests/extensions-compatibility.po.ts

```diff
@@ -0,0 +1,98 @@
+import PagePo from '@/cypress/e2e/po/pages/page.po';
+import { InstallChartPage } from '@/cypress/e2e/po/pages/explorer/charts/install-charts.po';
+import Kubectl from '@/cypress/e2e/po/components/kubectl.po';
+import ChartInstalledAppsPagePo from '@/cypress/e2e/po/pages/chart-installed-apps.po';
+import ProductNavPo from '@/cypress/e2e/po/side-bars/product-side-nav.po';
+import BaseResourceList from '@/cypress/e2e/po/lists/base-resource-list.po';
+import CodeMirrorPo from '@/cypress/e2e/po/components/code-mirror.po';
+import AsyncButtonPo from '@/cypress/e2e/po/components/async-button.po';
+import LabeledInputPo from '@/cypress/e2e/po/components/labeled-input.po';
+import CheckboxInputPo from '@/cypress/e2e/po/components/checkbox-input.po';
+
+const installChart = new InstallChartPage();
+const terminal = new Kubectl();
+const installedApps = new ChartInstalledAppsPagePo();
+
+export default class ExtensionsCompatibiliyPo extends PagePo {
+  title(selector: string): Cypress.Chainable<string> {
+    return this.self().find(selector).invoke('text');
+  }
+
+  waitForTitle(selector: string, title: string) {
+    return this.title(selector).should('contain', title);
+  }
+
+  waitForInstallChartPage(repoName:string, chartName:string) {
+    return installChart.waitForChartPage(repoName, chartName);
+  }
+
+  chartInstallNext() {
+    return installChart.nextPage();
+  }
+
+  chartInstallClick() {
+    return installChart.installChart();
+  }
+
+  chartInstallWaitForInstallationAndCloseTerminal(interceptName: string, installableParts: Array<String>) {
```

Comment:
comment only. this one adds max 25 seconds on every run. given though these only run nightly maybe they don't have to be optimised

Reply:
Optimisation is interesting because there's stuff that I check for but propagation of data on the backend to get stuff working is not something I can really track from the UI and there are several examples of that on the backend parts of extensions like Elemental and especially Kubewarden.

There's parts where we can only do so much, hence I've added quite a few `cy.wait` on the code in key areas. Since it runs nightly, time optimisation isn't a big factor at this point, but I've tried to avoid `cy.wait` as much as I could.

Extra comment: this method has been refactored in https://github.com/rancher/dashboard/pull/11626/files to allow to pass custom timeouts. KW is a different beast 😛